databricks_cli/unity_catalog/metastore_cli.py

```diff
@@ -0,0 +1,218 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import click
+
+from databricks_cli.click_types import MetastoreIdClickType, WorkspaceIdClickType, JsonClickType
+from databricks_cli.configure.config import provide_api_client, profile_option, debug_option
+from databricks_cli.unity_catalog.utils import mc_pretty_format
+from databricks_cli.unity_catalog.api import UnityCatalogApi
+from databricks_cli.utils import eat_exceptions, CONTEXT_SETTINGS, json_cli_base
+
+
+#################  Metastore Commands  #####################
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Create a metastore.')
+@click.option('--name', required=True, help='Name of the new metastore.')
+@click.option('--storage-root', required=True,
+              help='Storage root URL for the new metastore.')
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def create_metastore_cli(api_client, name, storage_root):
+    """
+    Create new metastore specified by the JSON input.
+
+    Calls the 'createMetastore' RPC endpoint of the Unity Catalog service.
+    Returns the properties of the newly-created metastore.
+
+    """
+    metastore_json = UnityCatalogApi(api_client).create_metastore(name, storage_root)
+    click.echo(mc_pretty_format(metastore_json))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='List metastores.')
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def list_metastores_cli(api_client):
+    """
+    List metastores.
+
+    Calls the 'listMetastores' RPC endpoint of the Unity Catalog service.
+    Returns array of MetastoreInfos.
+
+    """
+    metastores_json = UnityCatalogApi(api_client).list_metastores()
+    click.echo(mc_pretty_format(metastores_json))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Get a metastore.')
+@click.option('--id', 'metastore_id', required=True, type=MetastoreIdClickType(),
+              help='Unique identifier of the metastore to get.')
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def get_metastore_cli(api_client, metastore_id):
+    """
+    Get a metastore.
+
+    Calls the 'getMetastore' RPC endpoint of the Unity Catalog service.
+    Returns nothing.
+
+    """
+    metastore_json = UnityCatalogApi(api_client).get_metastore(metastore_id)
+    click.echo(mc_pretty_format(metastore_json))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Update a metastore.')
+@click.option('--id', 'metastore_id', required=True, type=MetastoreIdClickType(),
+              help='Unique identifier of the metastore to update.')
+@click.option('--json-file', default=None, type=click.Path(),
```

Comment:
i'm not sure if abundance of `--json-file` options is a good sign for CLI UX. every parameter within JSON should become own argument, unless there's a complex type, like cluster or job settings.

Reply:
Agreed.  Tricky part is that many of the UC objects are fairly complex types.  Metastore probably qualifies.